roles/network_plugin/canal/templates/cni-canal.conflist.j2

```diff
@@ -15,7 +15,11 @@
           "type": "k8s"
         },
         "kubernetes": {
+        {% if kubeadm_enabled %}
+          "kubeconfig": "{{ kube_config_dir }}/kubelet.conf"
+        {% else %}
           "kubeconfig": "{{ kube_config_dir }}/node-kubeconfig.yaml"
```

Comment:
Would it be smarter to just rename the `kubelet.conf` to `kubelet.conf` so it matches?

Reply:
That was my first thought, but that might break clusters set up in standard mode. They might depend on the `node-kubeconfig.yaml` name. A better approach might be to use a variable to hold the name of the file, but that would not play nice with how the files are written in `roles/kubernetes/node/tasks/main.yml`, and be a much more intrusive change. IMO this is the easiest and most compatible way to fix this.

Reply:
if `{{ kube_config_dir }}/node-kubeconfig.yaml` exists, symlink `{{ kube_config_dir }}/kubelet.conf` to it?

Reply:
Agreed, in the standard case we could create the symlink, and let the cni-*.conflist.j2 reference `kubelet.conf`. It will let the two modes converge gracefully.
I will update the PR.